bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/AbstractAsyncBundleProcessor.java

```diff
@@ -107,16 +107,16 @@ protected boolean isBundleRelevant(Bundle bundle) {
     }
 
     /**
-     * Checks for the existence of a given directors inside the bundle.
+     * Checks for the existence of a given directors inside the bundle and its attached fragments.
      *
      * Helper method which can be used in {@link #isBundleRelevant(Bundle)}.
      * 
      * @param bundle
      * @param path the directory name to look for
-     * @return <code>true</code> id the bundle contains the given directory
+     * @return <code>true</code> id the bundle or one of its attached fragments contain the given directory
      */
     protected final boolean isDirectoryPresent(Bundle bundle, String path) {
-        return bundle.getEntry(path) != null;
+        return bundle.findEntries(path, "*", true) != null;
```

Comment:
> Now you confused me a little bit. AFAIK this method is only used for parsing the ESH-INF subdirectories, whether they are available or not. It is called with e.g. "ESH-INF/config" and should prove whether the directory is there or not.

I implemented my test using a normal OSGi bundle without using the Eclipse SmartHome framework, a ESH-INF directory, etc. Sorry for the confusion, I have tested it using META-INF because that directory is present for all OSGi bundles.

> Previously this was done with getEntry(path which does not work for fragments.

I am fine with supporting fragments, so it is okay to use another function that look up fragments, too.

> Where's the problem with my apporach?

Let's forget about fragment at the moment, because using the "findEntries" function is fine for me.

We want to check if a directory is present.
Your code change will call findEntries in a way, that it builds a list of all entries below a path (you set recursive and using a wildcard for filePattern). This wastes memory, I/O, CPU, etc.
If you want to know if there is a directory "Program Files" on the drive C: of your windows machine, I assume you will not start to write down all files / directories that are present in that directory and the sub-directories (etc. / recursive), just to state "if there is at least one entry, the directory is present".

The Eclipse SmartHome framework is used by different products in very different ways.
Let's assume I have x bundles that contains a lot of thing type XMLs (the number of thing XML could be very big [e.g. thousands] if you want to support a wide range of device types, e.g. the whole Z-Wave database).
You should a very very small embedded device with very little RAM now build a list of all that files, just to throw the list away after we see if there is one?

IIRC the function is called every time a bundle gets started, so if we start / stop bundles multiple times, this will lead in doing this factorized approach every time.

So, if there is a way, to check for the directory without using a wildcard and a recursive search, we should invest the time to implement this once and not waste time every time a bundle gets started.

Do you understand my problem now?